jormungandr/src/settings/logging.rs

```diff
@@ -16,7 +16,12 @@ use tracing_subscriber::fmt::SubscriberBuilder;
 use tracing_subscriber::layer::SubscriberExt;
 use tracing_subscriber::layer::{Layer, Layered};
 
-pub struct LogSettings(pub Vec<LogSettingsEntry>);
+pub struct LogSettings(pub Vec<LogSettingsEntry>, pub LogInfoMsg);
+
+/// A wrapper to return an optional string message that we
+/// have to manually log with `info!`, we need this because
+/// some code executes before the logs are initialized.
+pub type LogInfoMsg = Option<String>;
```

Comment:
I think this way is more self-descriptive and does not require a type alias (OTOH you use `LogInfoMsg` on its own in other places):

```suggestion
pub struct LogSettings {
    /// Configured log outputs.
    pub entries: Vec<LogSettingsEntry>,
    /// An optional string message that should be logged after the logs are initialized.
    pub override_warning: Option<String>,
}

```